### PR TITLE
Refactor/spin cssinjs

### PR DESCRIPTION
**Pull request by jrr997**
[[中文版模板 / Chinese template](https://github.com/ant-design/ant-design/blob/master/.github/PULL_REQUEST_TEMPLATE/pr_cn.md)]

### 🤔 This is a ...

- [ ] New feature
- [ ] Bug fix
- [ ] Site / documentation update
- [ ] Demo update
- [ ] Component style update
- [ ] TypeScript definition update
- [ ] Bundle size optimization
- [ ] Performance optimization
- [ ] Enhancement feature
- [ ] Internationalization
- [x] Refactoring
- [ ] Code style optimization
- [ ] Test Case
- [ ] Branch merge
- [ ] Other (about what?)

### 👀 Preview 

https://preview-34653-ant-design.surge.sh/components/spin-cn/

### 📝 Changelog

| Language   | Changelog |
| ---------- | --------- |
| 🇺🇸 English | Spin support cssinjs |
| 🇨🇳 Chinese | Spin 组件支持 cssinjs |

### ☑️ Self Check before Merge

⚠️ Please check all items below before review. ⚠️

- [x] Doc is updated/provided or not needed
- [x] Demo is updated/provided or not needed
- [x] TypeScript definition is updated/provided or not needed
- [x] Changelog is provided or not needed

### CODE REVIEW

**Comment by jrr997** [comment]
update token name

File: components/list/__tests__/__snapshots__/Item.test.js.snap [hunk 1]
```diff
@@ -82,44 +82,50 @@ exports[`List Item Layout rowKey could be function 1`] = `
     className="ant-list ant-list-split"
   >
     <Spin
```
Comment (zombieJ) [inline components/list/__tests__/__snapshots__/Item.test.js.snap:84]: 蛋疼，居然有个结构 diff。这边需要给 master 发个 PR，替换一下：
```tsx
-- expect(wrapper).toMatchSnapshot();
++ expect(wrapper.render()).toMatchSnapshot();
```

Comment (zombieJ) [reply components/list/__tests__/__snapshots__/Item.test.js.snap:84]: snapshot 需要保持一致

File: components/spin/style/index.tsx [hunk 2]
```diff
@@ -1,2 +1,245 @@
...
+const genSpinStyle: GenerateStyle<SpinToken> = (token: SpinToken, hashId: string): CSSObject => ({
+  [`${token.spinCls}`]: {
+    ...resetComponent(token),
+    position: 'absolute',
+    display: 'none',
+    color: token.primaryColors,
+    textAlign: 'center',
+    verticalAlign: 'middle',
+    opacity: 0,
+    transition: `transform 0.3s ${token.motionEaseInOutCirc}`,
```
Comment (zombieJ) [inline components/spin/style/index.tsx:37]: `0.3s` 用 token 里的值代替

File: components/spin/style/index.tsx [hunk 3]
```diff
@@ -1,2 +1,245 @@
...
+      display: 'inline-block',
+      opacity: 1,
+    },
+
+    '&-nested-loading': {
+      position: 'relative',
+      [`> div > ${token.spinCls}`]: {
+        position: 'absolute',
+        top: 0,
+        left: 0,
```
Comment (zombieJ) [inline components/spin/style/index.tsx:50]: left 使用 逻辑位置 inset 代替

File: components/spin/style/index.tsx [hunk 4]
```diff
@@ -1,2 +1,245 @@
...
+        zIndex: 4,
+        display: 'block',
+        width: '100%',
+        height: '100%',
+        maxHeight: 400,
```
Comment (zombieJ) [inline components/spin/style/index.tsx:55]: 数字硬编码加个 `// FIXME: hard code in v4`

File: components/spin/style/index.tsx (hunk 3 above)
Comment (zombieJ) [reply components/spin/style/index.tsx:50]: 下同，所有 left right 的都使用 逻辑位置代替，包括 margin padding border

File: components/spin/style/index.tsx (hunk 4 above)
Comment (zombieJ) [reply components/spin/style/index.tsx:55]: 下同

File: components/spin/style/index.tsx (hunk 2 above)
Comment (zombieJ) [reply components/spin/style/index.tsx:37]: 下同

File: components/spin/__tests__/__snapshots__/index.test.js.snap [hunk 5]
```diff
@@ -3,15 +3,21 @@
...
-  size="default"
-  spinning={true}
-  wrapperClassName=""
 >
-  <div
-    aria-busy={true}
-    aria-live="polite"
-    className="ant-spin ant-spin-spinning"
-  />
+  <Spin
```
Comment (zombieJ) [inline components/spin/__tests__/__snapshots__/index.test.js.snap:7]: 这边需要到 master 里发个 PR 把 snapshot 更新一下：
```diff
-- expect(wrapper).toMatchSnapshot();
++ expect(wrapper.render()).toMatchSnapshot();
```

**Pull request merged**